src/cpp/client/generic_stub.cc

```diff
@@ -59,6 +59,10 @@ GenericStub::PrepareUnaryCall(grpc::ClientContext* context,
                               const grpc::string& method,
                               const grpc::ByteBuffer& request,
                               CompletionQueue* cq) {
+  // Unary rpcs must send a payload
+  if (!request.Valid()) {
+    return nullptr;
```

Comment:
So I think that none of our calls can take nullptr as a result; we've never said that that is a possible outcome. If this is an API misuse, we may want to assert as we do in other API misuses.

As an additional note: we also have generic stub callback API versions that are implemented in the same file and tested in client_callback_end2end_test

Reply:
Let me expand. I think it looks like it's a misuse for _any_ send to be an invalid byte buffer, not just a unary call, is that correct? If so, would it make more sense for the assertion to go in the part of call_op_set.h that deals with send messages in general?

Reply:
@vjpai Originally, I thought it was only a misuse to send an invalid byte buffer in the unary case, since streaming is allowed to send "no message" and an invalid byte buffer is equivalent to "no message". But your comment prompted me to think about how it should probably be invalid to do a streaming Write with no message... so yeah, that would imply that we should assert that we're not sending invalid byte buffers for Send messages in general.

However, I think I need to discuss with you further whether that responsibility should be in call_op_set.h or elsewhere. Currently, we have:

https://github.com/grpc/grpc/blob/bebd20b126303a1f5d6cbb31d0463460b59c5074/include/grpcpp/impl/codegen/call_op_set.h#L312-L313

The invalidity of the ByteBuffer is currently part of how we determine if a Send op is needed.

Reply:
If my memory is correct, we use the nullptr to indicate that it is an invalid buffer (for example in read), but invalid buffer does not mean "no message". You can send an empty message with an empty buffer, but it should be a valid buffer with an empty slice.

Reply:
@yang-g thanks for correcting my incorrect statement. Do you have an opinion on where we should assert / return an error if a user tries to send an invalid buffer?

Reply:
I am fine with asserting at the points where an invalid buffer is used. Maybe add a comment in the ByteBuffer header to call out the empty buffer case.

Reply:
Done.